devdoc/twodarray_requirements.md

```diff
@@ -0,0 +1,168 @@
+# `two_d_array` requirements
+============================
+
+## Overview
+
+`TWO_D_ARRAY` is a module provides a 2D array with variable size of rows and columns. Each row has the same size to ensure easier access. The first dimension of array is resizable, it contains the pointer to the second dimension of array. The pointer is initialized to NULL and points to the allocated memory address when there's not enough space. When the memory stored in a row is not needed any more, the entire row shall be freed and set back to `NULL`.
```

Comment:
>shall be [](http://example.com/codeflow?start=419&length=8)

Nitpick: we use shall in the specs below. shall is very imperative, like must. In the overview of a module we usually describe in looser terms what the module does (using is, should, etc.). So I would say here "the entire row is freed"

Reply:
got the idea, changed to is